lib/wasabi/resolver.rb

```diff
@@ -41,10 +41,10 @@ def resolve
     private
 
     def load_from_remote
-      request.url = document
-      response = HTTPI.get(request, adapter)
+      request.adapter *adapter if adapter
+      response = request.get(document)
 
-      raise HTTPError.new("Error: #{response.code} for url #{request.url}", response) if response.error?
+      raise HTTPError.new("Error: #{response.status} for url #{document}", response) unless response.success?
```

Comment:
this uses the `Struct.new` syntax [introduced in ruby 3.2](https://bugs.ruby-lang.org/issues/16806), so it needs to be refactored to use the old style for compatibility with older rubies